### ISSUE TITLE
Update Dataproc and PySpark in Genetics

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/issues#3189]
Dataproc's image 2.2 has been public for one month and it comes with [PySpark 3.5](https://spark.apache.org/releases/spark-release-3-5-0.html#highlights). They have important features like support for all SQL functions and a basic testing suite.

This is a nice to have.

**Comment by ireneisdoomed** [comment]
Currently blocked. Hail does not support PySpark>3.4.

**Comment by ireneisdoomed** [comment]
Now unblocked. Hail is now compatible with Pyspark 3.5 https://github.com/hail-is/hail/pull/14158
